conda_build_all/builder.py

```diff
@@ -210,16 +211,21 @@ def compute_build_distros(self, index, recipes):
         return all_distros
 
     def main(self):
-        # We set the CONDA_NPY and CONDA_PY variables to meaningless values.
-        # Since we compute a build matrix anyway, these are only useful to
-        # prevent conda-build throwing a wobbly for them missing.
-        import conda_build.config
-        conda_build.config.config.CONDA_NPY = 10
-        conda_build.config.config.CONDA_PY = 10
-
-        recipe_metas = self.fetch_all_metas()
         index = get_index(use_cache=True)
 
+        # If it is not already defined with environment variables, we set the CONDA_NPY
+        # to the latest possible value. Since we compute a build matrix anyway, this is 
+        # useful to prevent conda-build bailing if the recipe depends on it (e.g.
+        # ``numpy x.x``), and to ensure that recipes that don't care which version they want
+        # at build/test time get a sensible version.
+        if conda_build.config.config.CONDA_NPY is None:
```

Comment:
For my own sake I want to make this statement:

In order to be able to parse all recipes, we **must** have a CONDA_NPY with the current implementation of conda_build. This is because recipes with a ``numpy x.x`` have a hard requirement on it being set. However, if we set it, we lose conda_build's useful capability of picking the latest numpy for situations where the requirement is simply ``numpy`` (i.e. not ``numpy x.x``). It is for this reason alone that this code block exists.